scripts/reset_iks_api_key.sh

```diff
@@ -48,18 +48,25 @@ fetch_data() {
 
     while [ "$url" != "null" ]; do
         # Fetch data from the API
-        response=$(curl -s "$url" --header "Authorization: $IAM_TOKEN" --header "Content-Type: application/json")
-
-        # Extract next URL and current data
-        next_url=$(echo "$response" | jq -r '.next')
-        key_descriptions=$(echo "$response" | jq -r --arg name "${APIKEY_KEY_NAME}" '.apikeys | .[] | select(.name == $name) | .description')
-        for i in "${key_descriptions[@]}"; do
-            if [[ "$i" =~ ${REGION} ]] && [[ "$i" =~ ${RESOURCE_GROUP_ID} ]]; then
-                echo "Found key named ${APIKEY_KEY_NAME} which covers clusters in ${REGION} and resource group ID ${RESOURCE_GROUP_ID}"
-                reset=false
-                break
-            fi
-        done
+        response=$(curl -s -w "%{http_code}" "$url" --header "Authorization: $IAM_TOKEN" --header "Content-Type: application/json")
+        status_code="${response: -3}"
+        response_body="${response%???}"
```

Comment:
this is messy - can we use something like this logic? https://github.ibm.com/GoldenEye/operations/blob/2eca29581538c709adc8bf7ce0f52b69251ef652/catalog-onboard/scripts/perform-sccv2-scan.sh#L158-L171